src/Endatix.Infrastructure/Integrations/Slack/SlackClient.cs

```diff
@@ -19,28 +21,64 @@ public class SlackClient : INotificationHandler<SubmissionCompletedEvent>, ISlac
+    {
         _formRepository = formRepository;
         _logger = logger;
         _slackSettings = options.Value;
+        _serviceProvider = serviceProvider;
     }
 
-    public async Task Handle(SubmissionCompletedEvent notification, CancellationToken cancellationToken) {
-        if(notification.Submission.IsComplete && _slackSettings.Active){
+    public async Task Handle(SubmissionCompletedEvent notification, CancellationToken cancellationToken)
+    {
+        Guard.Against.Null(_slackSettings.Active, nameof(_slackSettings.Active), "Slack settings must specify whether the integration is active.");
+        Guard.Against.NullOrEmpty(_slackSettings.EndatixHubBaseUrl, nameof(_slackSettings.EndatixHubBaseUrl), "EndatixHubBaseUrl must have a value within the Slack settings collection.");
+        Guard.Against.NullOrEmpty(_slackSettings.Token, nameof(_slackSettings.Token), "Token must have a value within the Slack settings collection.");
+        Guard.Against.NullOrEmpty(_slackSettings.ChannelId, nameof(_slackSettings.ChannelId), "ChannelId must have a value within the Slack settings collection.");
+
+        if (notification.Submission.IsComplete && (bool)_slackSettings.Active)
+        {
+
+            var submissionUrl = string.Format(SLACK_SUBMISSIONS_URL_TEMPLATE, _slackSettings.EndatixHubBaseUrl.TrimEnd('\\', '/'), notification.Submission.FormId);
+            Form? form;
+
+            using (var scope = _serviceProvider.CreateScope())
+            {
+                var repository = scope.ServiceProvider.GetRequiredService<IRepository<Form>>();
+                form = await repository.GetByIdAsync(notification.Submission.FormId, cancellationToken);
+            }
 
-            var submissionUrl = $"{_slackSettings.EndatixHubBaseUrl.TrimEnd('\\', '/')}/forms/submissions/{notification.Submission.FormId}";
-            
-            //TODO: Troubleshoot _formRepository being disposed below and uncomment the line in order to reference the form name correctly
-            // var form = await _formRepository.GetByIdAsync(notification.Submission.FormId, cancellationToken) ?? throw new Exception("Form not found by the FormId of the submission");
+            Guard.Against.Null(form, $"Unable to load submission {notification.Submission.Id}'s Form object.");
 
-            var message = $":page_with_curl: <{submissionUrl}|New submission> for form with id {notification.Submission.FormId}";
+            var message = string.Format(SLACK_MESSAGE_TEMPLATE, submissionUrl, form.Name);
 
             await PostMessageAsync(_slackSettings.Token, _slackSettings.ChannelId, message);
         }
+        else
+        {
+            if (!notification.Submission.IsComplete)
+            {
+                _logger.LogWarning($"SubmissionCompletedEvent raised on an incomplete submission with id {notification.Submission.Id}");
+            }
+
+        }
     }
 
     public async Task PostMessageAsync(string token, string channelId, string message)
```

Comment:
I have one more suggestion here. This method looks like a good candidate to become private. It seems it's only for internal use in the class (correct me if I'm wrong).
Something else I realize now - `token` and `channelId` are passed to the method but they are actually directly accessible from here as coming from a class member - `_slackSettings.Token` and `_slackSettings.ChannelId`. They can be removed from the member's input parameters.

Reply:
Done